atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java

```diff
@@ -94,13 +95,26 @@ public BackgroundSweeperImpl(LockAwareTransactionManager txManager,
                                  Supplier<Long> sweepPauseMillis,
                                  Supplier<Integer> sweepBatchSize,
                                  SweepTableFactory tableFactory) {
+        this(txManager, kvs, sweepRunner, isSweepEnabled, sweepPauseMillis, sweepBatchSize, tableFactory,
+                Optional.absent());
+    }
+
+    public BackgroundSweeperImpl(LockAwareTransactionManager txManager,
```

Comment:
We could consider doing the "`@Deprecate` and remove in the next release" thing by default, where the thing we want to break isn't actually harmful. I haven't read the PR in detail, but this sounds like it would not be harmful.